tests/endpoints/test_forms.py

```diff
@@ -41,3 +45,121 @@ def test_get__ok(self):
                     form_id=fixture["response_data"][0]["xmlFormId"]
                 )
                 self.assertIsInstance(observed, Form)
+
+    def test_update__requires_definition_or_attachments(self):
```

Comment:
I think the idiomatic way is self.assertRaises() context manager around the call that might raise, then check the exception args for the error message.